File: Data Structures & Algorithms/top-k-elements-in-list/submission-3.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        // 1. Build frequency map
        Map<Integer, Integer> fmap = new HashMap<>();
        for (int num : nums) {
            fmap.put(num, fmap.getOrDefault(num, 0) + 1);
        }

        // 2. Min-Heap: ordered by frequency ascending
        // (a, b) -> fmap.get(a) - fmap.get(b)
        PriorityQueue<Integer> minHeap = new PriorityQueue<>((a, b) -> fmap.get(a) - fmap.get(b));

        for (int key : fmap.keySet()) {
            minHeap.offer(key);
            // If heap size exceeds k, remove the element with the lowest frequency
            if (minHeap.size() > k) {
                minHeap.poll();
            }
        }

        // 3. Extract elements (they will be the k most frequent)
        int[] result = new int[k];
        for (int i = 0; i < k; i++) {
            result[i] = minHeap.poll();
        }

        return result;
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        st = set(nums)
        ans = 0

        for num in st:
            if num - 1 not in st:
                curr = num
                length = 1

                while curr + 1 in st:
                    curr += 1
                    length += 1

                ans = max(ans, length)

        return ans